support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -43,19 +44,43 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
   ): FutureHandlerResult = {
     val now = () => DateTime.now(DateTimeZone.UTC)
     val today = () => LocalDate.now(DateTimeZone.UTC)
+    val promotionService = services.promotionService
+    val redemptionService = services.redemptionService
+    val zuoraService = services.zuoraService
+    val isTestUser = state.user.isTestUser
+    val config: ZuoraConfig = zuoraConfigProvider.get(isTestUser)
+    doStepsFromDeps(state, requestInfo, now, today, promotionService, redemptionService, zuoraService, config)
+  }
+}
+object CreateZuoraSubscription {
+  import com.gu.FutureLogging._
+
+  def doStepsFromDeps(
+    state: CreateZuoraSubscriptionState,
+    requestInfo: RequestInfo,
+    now: () => DateTime,
+    today: () => LocalDate,
+    promotionService: PromotionService,
+    redemptionService: DynamoLookup with DynamoUpdate,
+    zuoraService: ZuoraSubscribeService,
+    config: ZuoraConfig
+  ): Future[HandlerResult[SendThankYouEmailState]] = {
     for {
-      subscriptionData <- buildSubscriptionData(state, services.promotionService, GetCodeStatus.withDynamoLookup(services.redemptionService), today)
+      subscriptionData <- buildSubscriptionData(state, promotionService, GetCodeStatus.withDynamoLookup(redemptionService), today, config)
       subscribeItem = buildSubscribeItem(state, subscriptionData)
       identityId <- Future.fromTry(IdentityId(state.user.id))
         .withLogging("identity id")
-      maybeDomainSubscription <- GetSubscriptionWithCurrentRequestId(services.zuoraService, state.requestId, identityId, state.product.billingPeriod, now)
-          .withLogging("GetSubscriptionWithCurrentRequestId")
-      previewPaymentSchedule <- PreviewPaymentSchedule(subscribeItem, state.product.billingPeriod, services, checkSingleResponse)
-          .withLogging("PreviewPaymentSchedule")
+      maybeDomainSubscription <- GetSubscriptionWithCurrentRequestId(zuoraService, state.requestId, identityId, state.product.billingPeriod, now)
+        .withLogging("GetSubscriptionWithCurrentRequestId")
+      paymentMethodWithPaymentSchedule <-
+        state.paymentMethod.leftMap(pm =>
+          PreviewPaymentSchedule(subscribeItem, state.product.billingPeriod, zuoraService, checkSingleResponse)
```

Comment:
we only need to get the payment schedule for paid subs, this should make corporate subs execute quicker.